### PR TITLE
Added support for new python front end to LBANN. 

### PR DESCRIPTION
**Pull request by bvanessen**
Created new packages for necessary python packages.

### CODE REVIEW

File: var/spack/repos/builtin/packages/cudnn/package.py [hunk 1]
```diff
@@ -12,6 +12,22 @@ class Cudnn(Package):
...
 
+    version('7.5.1-10.1-x64', '24017f4a56159d48fd5a31c8a930167b',
+            url='https://developer.download.nvidia.com/compute/redist/cudnn/v7.5.1/cudnn-10.1-linux-x64-v7.5.1.10.tgz')
+    version('7.5.1-10.1-ppc64le', '',
+            url='https://developer.download.nvidia.com/compute/redist/cudnn/v7.5.1/cudnn-10.1-linux-ppc64le-v7.5.1.10.tgz')
+    version('7.5.1-10.0-x64', '703ed4be4d242ff4bc0ca48aaf2029bc',
+            url='https://developer.download.nvidia.com/compute/redist/cudnn/v7.5.1/cudnn-10.0-linux-x64-v7.5.1.10.tgz')
+    version('7.5.1-10.0-ppc64le', '8348cbab01970b75836d945d8a75915e',
+            url='https://developer.download.nvidia.com/compute/redist/cudnn/v7.5.1/cudnn-10.0-linux-ppc64le-v7.5.1.10.tgz')
+    version('7.5-10.1-x64', '4edc7fcfeada9f2363e97f9875154e8f',
```
Comment (becker33) [inline var/spack/repos/builtin/packages/cudnn/package.py]: These should be 7.5.0. I know the version scheme here is awful, but we can make it slightly more consistently awful.

**Comment by becker33** [comment]
@alalazo this seems related to #11206

**Comment by bvanessen** [comment]
@becker33 @tgamblin @scheibelp I need help with the cudnn package.  For the x86_64 packages, they tarball uncompresses into cuda/[include | lib], but for the ppc64le packages the untar path is cuda/targets/ppc64le-linux/[include | lib].  How do I modify the recipe so that either they get untar'ed to the right place or we modify the spack LD_LIBRARY_PATH for the module?

File: var/spack/repos/builtin/packages/lbann/package.py [hunk 2]
```diff
@@ -81,18 +84,32 @@ class Lbann(CMakePackage):
...
                '~videostab ~videoio ~vtk', when='+opencv')
 
-    depends_on('protobuf@3.0.2: build_type=Release')
+    depends_on('protobuf@3.6.1: build_type=Release')
     depends_on('cnpy')
     depends_on('nccl', when='+gpu +nccl')
 
-    depends_on('conduit@master +hdf5', when='+conduit')
+    depends_on('conduit@master +hdf5', when='@0.94:0.99 +conduit')
+    depends_on('conduit@master +hdf5', when='@:0.90,0.99:')
```
[the remark this answers is not in the document]
Comment (bvanessen) [reply var/spack/repos/builtin/packages/lbann/package.py]: Done.

**Pull request merged**